### PR TITLE
Feature/build anim instance for setdress

### PR DESCRIPTION
**Pull request by kaamaurice**
## Changelog Description
- Build_anim create instances for setdress container using collection root datablock.
- anim instance accept collection type
- extract anim blend manage collections

### CODE REVIEW

**Comment by kaamaurice** [comment]
Oups missing this feature:
**Disabled by default if no setdress object animated**

File: openpype/hosts/blender/scripts/build_workfile.py [hunk 1]
```diff
@@ -747,18 +750,59 @@ def build_anim(project_name, asset_name):
...
+            asset_name=asset_name,
+            subset_name=f"animation{variant_name}",
+            datapath=datablock_type,
+            datablock_name=first_datablock.name,
+            use_selection=False,
+        )
+        animation_instance = bpy.context.scene.openpype_instances[-1]
+        if objects:
+            add_datablocks_to_container(objects, animation_instance)
+        if datablock_type == "collections":
```
Comment (Tilix4) [inline openpype/hosts/blender/scripts/build_workfile.py]: Please add comment

File: openpype/hosts/blender/scripts/build_workfile.py [hunk 2]
```diff
@@ -747,18 +750,59 @@ def build_anim(project_name, asset_name):
...
+        datablock_type = (
+            "collections"
+            if isinstance(first_datablock, bpy.types.Collection)
+            else "objects"
+        )
+        bpy.ops.scene.create_openpype_instance(
+            creator_name="CreateAnimation",
+            asset_name=asset_name,
+            subset_name=f"animation{variant_name}",
+            datapath=datablock_type,
```
Comment (Tilix4) [inline openpype/hosts/blender/scripts/build_workfile.py]: You better use `BL_TYPE_DATAPATH.get(type(first_datablock))`

Comment (Tilix4) [reply openpype/hosts/blender/scripts/build_workfile.py]: Also, can't you leave `datapath` and `datablock_name` empty and add to container all of them without pop the first one?

**Review by Tilix4 — APPROVED** [review]
Nice! One comment request and it's fine

**Review by Tilix4 — CHANGES_REQUESTED** [review]
Sorry, not approved as it is

File: openpype/hosts/blender/scripts/build_workfile.py (hunk 2 above)
Comment (kaamaurice) [reply openpype/hosts/blender/scripts/build_workfile.py]: I already try without datapath and datablock_name, get this result :
```python
RuntimeError: Error: Python: Traceback (most recent call last):
  File "D:\Kevin\DEV\OpenPype\openpype\hosts\blender\api\ops.py", line 603, in execute
    plugin.process(
  File "D:\Kevin\DEV\OpenPype\openpype\hosts\blender\api\plugin.py", line 378, in wrapper_exec_process
    return process_func(*args, **kwargs)
  File "D:\Kevin\DEV\OpenPype\openpype\hosts\blender\api\plugin.py", line 558, in process
    if op_instance.datablock_refs.get(d.name):
AttributeError: 'NoneType' object has no attribute 'name'
```

Comment (kaamaurice) [reply openpype/hosts/blender/scripts/build_workfile.py]: few changes but can't leave `datapath `and `datablock_name` empty.

**Review by Tilix4 — APPROVED** [review]
Perfect!

**Pull request merged**